base/log_keys.go

```diff
@@ -190,28 +216,32 @@ func ToLogKey(keysStr []string) (logKeys LogKey, warnings []DeferredLogFn) {
 	return logKeys, warnings
 }
 
-func inverselogKeyNames(in map[LogKey]string) map[string]LogKey {
+func inverselogKeyNames(in []string) map[string]LogKey {
 	var out = make(map[string]LogKey, len(in))
-	for k, v := range in {
-		out[v] = k
+	for i, v := range in {
```

Comment:
I'm on the fence about including some validation on the input slice in here, since we're iterating through it on init. A simple length check isn't enough for an array, as skipped values are counted in `len`.

It will allow us to catch when we've added a log key constant, but forgot to include it in the `logKeyNames` array.

```suggestion
	for i, v := range in {
		if v == "" && i != int(KeyNone) {
			panic(fmt.Sprintf("Empty value for logKeyNames[%d]", i))
		}
```

`panic: Empty value for logKeyNames[21]`

Reply:
Good idea - I had wondered about the best way to ensure we didn't omit anything during initialization of logKeyNames.